### DISCUSSION TITLE
Config Templates Feedback

### DISCUSSION THREAD

**Discussion by Jeffail** [redpanda-data/connect#785] [Ideas]
Hey everyone, similar to the Plugins V2 API there's some new experimental functionality called config templates that is ready to try. It's going to allow you to define parameterised templates for Benthos configuration snippets that can be imported with a cli flag and used like native Benthos components.

This is super useful in situations where you have commonly used configuration patterns but with small differences that prevent you from using the exact same config as a resource.

The current state of templates is that they'll be included in the next release as an experimental feature, meaning any aspect of this functionality is subject to change outside of major version releases. This includes the config spec of templates, how they work, and so on.

You can see some examples of what they look like in this directory: https://github.com/Jeffail/benthos/tree/master/template, and the (work in progress) documentation can be found at: https://github.com/Jeffail/benthos/blob/master/website/docs/configuration/templating.md

**Comment by mannharleen** [comment]
Just created my first template. The learning curve was steep, especially if one is new to benthos. 
Q. I found it simpler to create a template if I create the standard config.yaml using existing components first. So I ask, is it possible to create a cli subcommand utility that helps convert a resource into template? That will speed up template creation, and eventually help us create an ecosystem of higher level components (think: SaaS specific processors)?
